analyzers/src/SonarAnalyzer.CSharp/Rules/MessageTemplates/LoggingTemplatePlaceHoldersShouldBeInOrder.cs

```diff
@@ -0,0 +1,102 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.MessageTemplates;
+
+public sealed class LoggingTemplatePlaceHoldersShouldBeInOrder : IMessageTemplateCheck
```

Comment:
This is the last `IMessageTemplateCheck`, so we will not touch `MessageTemplateAnalyzer` afterwards.
Since the optimization PR has been merged, could you add a `CompilationStartAction` in the parent analyzer, that does something [similar to this](https://github.com/SonarSource/sonar-dotnet/blob/3d0ff4ce7fec7748d076ed869daf28ff15eaae5e/analyzers/src/SonarAnalyzer.CSharp/Rules/MessageTemplates/MessageTemplatesShouldBeCorrect.cs#L39)? 

It should look almost identical, since this rule also cares about `Microsoft`, `Serilog`, `NLog`.

Reply:
Also, just a reminder (I also forgot and just did it on my PRs): Please add an IT for this rule.